nixos/modules/programs/nano.nix

```diff
@@ -32,20 +33,15 @@ in
     };
   };
 
-  ###### implementation
+  config = lib.mkIf cfg.enable {
+    environment = {
+      etc.nanorc.text = lib.concatStringsSep "\n" cfg.nanorc;
+      systemPackages = [ cfg.package ];
+    };
 
-  config = lib.mkIf (cfg.nanorc != "" || cfg.syntaxHighlight) {
-    environment.etc.nanorc.text = lib.concatStringsSep LF (
-      ( lib.optionals cfg.syntaxHighlight [
-          "# The line below is added because value of programs.nano.syntaxHighlight is set to true"
-          ''include "${pkgs.nano}/share/nano/*.nanorc"''
-          ""
-      ])
-      ++ ( lib.optionals (cfg.nanorc != "") [
-        "# The lines below have been set from value of programs.nano.nanorc"
-        cfg.nanorc
-      ])
-    );
+    programs.nano.nanorc = lib.mkIf cfg.syntaxHighlight ''
+      # load syntax highlighting files
+      include "${cfg.package}/share/nano/*.nanorc"
```

Comment:
will this be at the beginning or the end
https://github.com/nixos/nixpkgs/commit/8ca0687b97298518ec7a074427de074d06ebc20a
it should be at the end

is lib.types.lines additive or will this override the config?

Reply:
explicitly moved it before

> is lib.types.lines additive or will this override the config?

yes, unless other priorities like mkForce are used.